types/mime-db/index.d.ts

```diff
@@ -1,21 +1,33 @@
 // Type definitions for mime-db 1.27
 // Project: https://github.com/jshttp/mime-db
 // Definitions by: AJP <https://github.com/AJamesPhillips>
+//                 Linus Unnebäck <https://github.com/LinusU>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 3.4
 
-/**
- * Data Structure
- * If unknown, every property could be undefined.
- * @see {@link https://github.com/jshttp/mime-db#data-structure}
- */
-export interface DataStructure {
-    /** where the mime type is defined. If not set, it's probably a custom media type. */
-    source?: string;
-    /** known extensions associated with this mime type. */
-    extensions?: string[];
-    /** whether a file of this type can be gzipped. */
-    compressible?: boolean;
-    /** the default charset associated with this type, if any. */
-    charset?: string;
+declare namespace database {
+    /**
+     * @see {@link https://github.com/jshttp/mime-db#data-structure}
+     */
+    interface MimeEntry {
+        /** Where the mime type is defined. If not set, it's probably a custom media type. */
+        readonly source?: string;
+        /** Known extensions associated with this mime type. */
+        readonly extensions?: readonly string[];
```

Comment:
You probably want to use `ReadonlyArray<string>` instead as that's more compatible.

Reply:
I wasn't aware that `ReadonlyArray` was available before `readonly ...[]`, thanks!